### PR TITLE
Fix Hanabi final-life termination timing

### LINKED ISSUE #179: [Bug] Hanabi game terminates an extra timestep after lives are lost
Line 427 in hanabi_game.py, I think it should be: `jnp.logical_or(out_of_lives, game_won), last_round_done`  NOT `jnp.logical_or(state.out_of_lives, game_won), last_round_done`? This makes the game terminates one timestep late.

### PR DESCRIPTION
**Pull request by ravihammond**
## Summary

This fixes a Hanabi termination bug where the environment could report episode completion one step late after the final life token was lost.

## Root cause

In `HanabiGame.step_game()`, the updated `out_of_lives` flag was computed from the post-action state, but terminal detection still used `state.out_of_lives` from the previous state. As a result, if the current move consumed the last remaining life token, `terminal` could remain false until the following step.

## Fix

- compute `terminal` using the updated `out_of_lives` value

## Tests

Added deterministic coverage for the exact failure mode:
- regression/unit test for `step_game()` terminal transition
- integration test for `step_env()` returning immediate done on final-life loss
- integration test for public `env.step()` autoreset behavior after the terminal transition

Also re-ran the generic Hanabi base API checks.

## Validation

- `.venv/bin/python -m pytest tests/hanabi/test_hanabi.py -q`
- `.venv/bin/python -m pytest tests/test_envs_implement_base_api.py -k hanabi -q`

Fixes #179

### CODE REVIEW

**Review by amacrutherford — APPROVED** [review]
lgtm
